src/main/java/com/example/ludogoriesoft/lukeriaerpapi/controllers/UserController.java

```diff
@@ -17,31 +18,37 @@
 public class UserController {
     private final UserService userService;
 
-
     @GetMapping
     public ResponseEntity<List<UserDTO>> getAllUsers() {
         return ResponseEntity.ok(userService.getAllUsers());
     }
 
     @GetMapping("/{id}")
-    public ResponseEntity<UserDTO> getUserById(@PathVariable(name = "id") Long id) {
-        return ResponseEntity.ok(userService.getUserById(id));
+    public ResponseEntity<UserDTO> getUserById(@PathVariable(name = "id") Long id) throws ChangeSetPersister.NotFoundException {
+            UserDTO user = userService.getUserById(id);
+            return ResponseEntity.ok(user);
     }
 
     @PostMapping
     public ResponseEntity<UserDTO> createUser(@Valid @RequestBody User user) {
```

Comment:
Would be better to use `UserDTO` as a parameter instead of User. In this case the validations should be moved from the model to the `UserDTO`

Reply:
In order to avoid having a password field in UserDTO, we submit User user for saving. In the new commit, I tried to have validations in UserDTO